handler.go

```diff
@@ -94,19 +94,49 @@ func isIPv6(addr string) bool {
 	return ip != nil && ip.To4() == nil
 }
 
+// A responseWriterWithFirstWrite is an http.Responsewrite with the ability to
+// tracks the time when the first response write happened.
+type responseWriterWithFirstWrite interface {
+	http.ResponseWriter
+	didWrite()
+	firstWrite() time.Time
+}
+
 // A timedResponseWriter tracks the time when the first response write
 // happened.
 type timedResponseWriter struct {
-	firstWrite time.Time
+	firstWriteTime time.Time
 	http.ResponseWriter
 }
 
+// A timedResponseWriterHijacker is a timedResponseWriter and http.Hijacker.
+type timedResponseWriterHijacker struct {
+	*timedResponseWriter
+	http.Hijacker
+}
+
+// NewLoggedResponseWriter wraps the provided http.ResponseWriter with Status
+// preserving the support to hijack the connection if supported by the provided
+// http.ResponseWriter.
+func newTimedResponseWriter(w http.ResponseWriter) responseWriterWithFirstWrite {
+	tw := &timedResponseWriter{ResponseWriter: w}
+	if hj, ok := w.(http.Hijacker); ok {
+		return &timedResponseWriterHijacker{timedResponseWriter: tw, Hijacker: hj}
+	}
+
+	return tw
+}
+
 func (w *timedResponseWriter) didWrite() {
-	if w.firstWrite.IsZero() {
-		w.firstWrite = time.Now()
+	if w.firstWriteTime.IsZero() {
+		w.firstWriteTime = time.Now()
 	}
 }
 
+func (w *timedResponseWriter) firstWrite() time.Time {
+	return w.firstWriteTime
+}
+
```

Comment:
this is not needed either, just make FirstWriteTime public

Reply:
shouldn't need to be public - it's only accessed in this file - IMO this getter function is just not necessary at all

Reply:
All related to the interface I added. Can't remember why at the moment but the only explanation I have us that it is required. Will check.